### ISSUE TITLE
'Sourcemap is likely to be incorrect' when running under Vite

### ISSUE THREAD

**Issue by Shaddix** [iendeavor/import-meta-env#9]
When running `vite build` I see the following in console:

> Sourcemap is likely to be incorrect: a plugin (import-meta-env) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help

Is it expected?
Are there real issues with sourcemaps after using the plugin?

P.S. I have sourcemaps turned on in config file:
```
  build: {
    sourcemap: true,
  },
```

**Comment by shamrt** [comment]
While not perfect, I was able to generate most sourcemaps using Rollup options:

```ts
  {
    build: {
      // https://rollupjs.org/guide/en/#big-list-of-options
      rollupOptions: {
        output: [{ sourcemap: true }],
      },
    },
  }
```

Would still be ideal if this plugin output its own sourcemaps for Vite's (esbuild's?) transformers…

**Comment by iendeavor** [comment]
Since environment variables are replaced after the build step, we cannot generate accurate source maps.

**Comment by hvanoch** [comment]
Is there no way to circumvent this issue?

**Comment by iendeavor** [comment]
Sadly, currently, no.

I did try to minimize the impact of source maps by using virtual modules, but this is a problem in some cases: https://github.com/BuilderIO/qwik/issues/1659. 😢 

Maybe we should expose the environment variable back to `globalThis.__ENV`, but it pollutes the global and can cause problems if we are building a micro-frontend application.

What do you think 😅 ?

**Comment by shamrt** [comment]
Could we build the sourcemaps in as a configurable option?

**Comment by iendeavor** [comment]
@shamrt Do you mean that even the sourcemap is wrong 🤔?

**Comment by iendeavor** [comment]
Source maps is released in v0.3.0 🚀 .

**Comment by bourquep** [comment]
I'm seeing this error with Vite 4.4.9 and 

![CleanShot 2023-10-01 at 11 14 08@2x](https://github.com/iendeavor/import-meta-env/assets/65096/7021411d-6294-4570-9395-c72b4b62e797)

**Comment by iendeavor** [comment]
Thank you for your report, this issue is fixed and published in `@import-meta-env/unplugin@0.4.10`. ❤️
